test-js/unit/pentaho/visual/base/modelFactory.Spec.js

```diff
@@ -0,0 +1,141 @@
+define([
+  "pentaho/type/Context",
+  "pentaho/visual/base"
+], function(Context, modelFactory) {
+  "use strict";
+
+  describe("pentaho/visual/base", function() {
+    var context;
+    var Model;
+    var dataSpec;
+
+    beforeEach(function() {
+      context = new Context();
+      Model = context.get(modelFactory);
+      var data = {
+        model: [
+          {name: "country", type: "string", label: "Country"},
+          {name: "sales", type: "number", label: "Sales"}
+        ],
+        rows: [
+          {c: [{v: "Portugal"}, {v: 12000}]},
+          {c: [{v: "Ireland"}, {v: 6000}]}
+        ]
+      };
+      dataSpec = {
+        v: data
+      };
+    });
+
+    it("can be instantiated with a well-formed spec", function() {
+      expect(function() {
+        return new Model({
+          width: 1,
+          height: 1,
+          interactive: false,
+          data: dataSpec
+        });
+      }).not.toThrowError();
+    });
+
+    it("can be instantiated without arguments", function() {
+      expect(function() {
+        return new Model();
+      }).not.toThrowError();
+    });
+
+    it("can be instantiated even if the modelSpec is not an object ", function() {
```

Comment:
Clearly, I dislike this unit test. It's not something you wanna test and set in stone (for us and our api's users) whether certain arbitrary values that explicitly go against the documented contract do not throw and result in being equivalent to a `nully` argument. This is circumstantial behaviour that can change with any rewrite of the code. The values `null` and `undefined` are expected to be equivalent to not passing any arguments. All the others are (for me) unwanted, undesired assertions.